rpc/provider/smoldot.ts

```diff
@@ -66,7 +66,13 @@ export const smoldotProvider: Provider<
       })()
     },
     release: async () => {
-      const { cleanUp, listeners, inner } = await connection(props, listener)
+      let conn
+      try {
+        conn = await connection(props, listener)
+      } catch (_error) {
+        return
```

Comment:
Although an error would likely mean that the connection is dead / "released", this––for some reason––seems not ideal. What're your thoughts on this?

Reply:
I believe that it would be better to just do
```
conn = connections.get(props)
```

Then, we need to decide what release should do if there is no connection.
One option could be that if there isn't anything to release, then do nothing.
Thoughts?